### ISSUE TITLE
 Access to the file revoked by the file manager

### ISSUE THREAD

**Issue by jvc2015** [Kunzisoft/KeePassDX#626]
Hi,

I make my file of database in root directory of my cellphone. They haved many passwords.
But, today don't can open my database why don't have access to the file by the file manager.
Opened my file manager ES File Explorer Pro, i don't exist my file of database anymore
Please help me to search the file, why i can't find It in any directory of my data storage, too Is possible find a programated task of cleaning un the file manager.
My cellphone Is Ulefone Amor 5S
Android version 9
KeePass version 2.7
File manager: ES File Manager v4.2.2.1

Sorry for my english, i speak spanish lenguaje.
Thanks for your help

**Comment by J-Jamet** [comment]
Hello, no worries about the English language, I am not an English speaker either.

Regarding your problem, I can't really help you. It's like telling me that you lost a file in the architecture of your hard drive. I have no way of knowing where it is.
KeePassXC is a file editor and not a file manager, so it does not know directly where the files are stored, it only knows the file URIs (it's a bit like URLs for websites, the browser does not don't know where the files are stored on the remote server, there are only requests and streams)
This allows KeePassDX to never have direct control over the file so as not to be in charge of disk operations.

I'm not familiar with ES File Explorer Pro but maybe only the URI was invalidated and your file is where you saved it the first time. Or maybe this app deleted the file somehow, I don't know.
I can only add information to the wiki about this file manager if I do more testing.

**Comment by whoizit** [comment]
long tap on the open button will open the file in the default file manager, but this does not solve this issue

**Comment by whoizit** [comment]
and no, I have another problem. I don't have a path to the database, I have to reopen it every time, cause ~'file access is cancelled by the file manager'. But the headline is exactly the same as the one I would write

**Comment by J-Jamet** [comment]
@whoizit You do not specify which file manager you are using. If it revokes the link every time, I'll add the you're app to the wiki: https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync

**Comment by whoizit** [comment]
default file manager `com.android.documentsui from crDroid Android 9 firmware
and over CX Explorer `com.cxinventor.file.explorer`

**Comment by J-Jamet** [comment]
In this case, I recommend that you use another file manager that does not cancel links during an ACTION_OPEN_DOCUMENT. Or request behavior change in CX Explorer.

**Comment by whoizit** [comment]
maybe `syncthing` changes the time of file change and it breaks?

**Comment by J-Jamet** [comment]
Yes that's a possibility, I don't use Syncthing. We have to test.

**Comment by whoizit** [comment]
after a long click to open, it opens in the default file manager and the path to the file is this:
`/document/primary:sync/keepassxc/passwords.kdbx`

If I close the KeePassDX, the path to the file is - `/root/storage/emulated/0/sync/keepassxc/passwords.kdbx` and _file access is cancelled by the file manager_

**Comment by J-Jamet** [comment]
The long click is only an alternative solution which uses the Intent `ACTION_GET_CONTENT. This is not the normal method used by SAF to open documents over the long term, so this is normal. It is a opening solution for old file managers which are not yet up to date.
Whenever possible, try to use the single click that calls `ACTION_OPEN_DOCUMENT` for native integration with file manager.

**Comment by metakaos** [comment]
> after a long tap to open button, it opens in the default file manager and the path to the file is this:
> `/document/primary:sync/keepassxc/passwords.kdbx`
> 
> If I close and reopen the KeePassDX, the path to the file is - `/root/storage/emulated/0/sync/keepassxc/passwords.kdbx` and _file access is cancelled by the file manager_

One workaround that's working for me so far was hitting the back arrow button next to that path once the database was open and setting that database as a favorite (star), so it opens it using  `/document/primary:sync/keepassxc/passwords.kdbx` every time instead of trying to use `/root/storage/emulated/0/sync/keepassxc/passwords.kdbx`

**Comment by whoizit** [comment]
It seems to be decided for me. I do not know how. At first it seems to have helped to remove Cx File Explorer, but now it works with it as well.

**Comment by michas0602** [comment]
Same (?) problem on 2 mobiles since last update - it tells me every time I open KeepassDX " file not found please open the file manager" and access refused by filemanager. Some weeks ago everything was fine, I am not 100% sure if the last update caused it, but it looks so.

Very annoying as I now have to close the warnings and reselect the keepass file every time.

OS on both mobiles is LOS 17.1.

**Comment by J-Jamet** [comment]
@michas0602 I am almost sure that the problem does not come from the update of KeePassDX since the SAF which manages the URIs of the databases has not been modified for a long time.
You haven't indicated which file manager you are using (and whether it has had an update).
The problem could also be a link that you favored with the star trying to be opened when it is broken.
Tried to put the star on the new link you select.

**Comment by michas0602** [comment]
I just dont know what else could cause this - especially on my second mobile. File manager is LOS-standard + Total Commander.
Another thing I've seen is that the message occurs if I open Keepass through MagiKeyboard and if I open the app. But why does it open my keepass-file normal through the app if I press back one time in the app and select the file (I have just one) without any error?

But when I open the app, it always says "/document/2160" and the error-message "access denied by filemanager" or something like this.

What exactly has the filemanager to do with this access? Is there a place for keepass-file where I can copy it fir a test where it definately should work?
Strange thing to me is that everything was fine here a few weeks ago - and I have not moved my keepass-file.

**Comment by michas0602** [comment]
To proof I just reinstalled older 2.9.1 from Github here - works like a charm, no problem anymore here on mobile 1...maybe I do more tests tomorrow, but as said - 2.9.1 does not make this problem here.

**Comment by J-Jamet** [comment]
> I just dont know what else could cause this - especially on my second mobile. File manager is LOS-standard + Total Commander.
Another thing I've seen is that the message occurs if I open Keepass through MagiKeyboard and if I open the app. But why does it open my keepass-file normal through the app if I press back one time in the app and select the file (I have just one) without any error?

I have already answered the question, you have stared a broken link. Please read the previous answer carefully.
You can see broken links in favorites with Settings -> App Settings -> Hide broken database links

> But when I open the app, it always says "/document/2160" and the error-message "access denied by filemanager" or something like this.

Because the database link you want to open has been broken by the file manager.

> What exactly has the filemanager to do with this access?

The File Manager is the app that provides the URI link to KeePassDX to retrieve the data from the file. 
https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync 
This wiki explains very well how URIs work

> Is there a place for keepass-file where I can copy it fir a test where it definately should work?

All locations authorized by your file manager which keeps links.

> Strange thing to me is that everything was fine here a few weeks ago - and I have not moved my keepass-file.

Your file manager has invalidated a link for reason X or Y (time, file modification, ...)

> To proof I just reinstalled older 2.9.1 from Github here - works like a charm, no problem anymore here on mobile 1...maybe I do more tests tomorrow, but as said - 2.9.1 does not make this problem here.

This is not proof at all, it just means you have a different link that works on another device. It is possible but tell me exactly what is the change in this version that is causing the problem.

**Comment by Unspec7** [comment]
Confirming that this is happening on Android 11 using the default file manager. Only happens when opening keepassdx from inside an app, i.e. using the keyboard option or the autofill option.

Has rendered this app completely unusable.

**Comment by guillp** [comment]
FYI I had the same issue on my OnePlus Nord and it looks like it has been fixed by an update of OnePlus FileManager in 29/12/2020.

**Comment by EthanWeegee** [comment]
I'm not sure what I changed but as of recently I'm having this issue too. I use Material Files which is said to work fine. May or may not be related but the app is now unusably slow (several seconds of pauses) when trying to open a database. Opening a database instantly says 'access revoked by the file manager'.

**Comment by pepa65** [comment]
> One workaround that's working for me so far was hitting the back arrow button next to that path once the database was open and setting that database as a favorite (star),

Yes, this worked when I encountered "Access to the file revoked by the file manager" all the time. Once it's starred, it's apparently OK to remember the path..?

**Comment by E0697** [comment]
This has been happening a lot recently! 
It was working perfectly before. I noticed a recent update of the Files app, which could be the cause here. When I encounter it, quiting KPDX and removing it from the recents menu and then launching it again fixes it.
Any thoughts?!

**Comment by horbo** [comment]
Hello! I have the same problem! Sometimes Keepass show the "Access to the file revoked by the file manager", don't know what triggers that. The file exists, and message still remains if I select the file manually in the app once again.

The message disappears only after restarting the phone or force-closing the app using Android settings menu.

I'm using Syncthing to sync files among my devices, but the problem started to occur maybe few weeks ago.

**Comment by E0697** [comment]
Hello!
Yes, that's exactly what's happening for me.
I'm using a local database and syncing it to the cloud.

**Comment by fevieira27** [comment]
The same issue has indeed happened to me today for the first time. I'm using Google files with the DB stored on drive.

For now, clicking on the back arrow and opening the DB in keepass DX again from scratch did solve the issue. It could have been due to a files update that happened 2 days ago (I'm on their beta channel), so hope the error doesn't come back.

**Comment by soredake** [comment]
This started to happen VERY often with keepassdx from gplay + google drive client.

**Comment by soredake** [comment]
This happen reliably after 2-4 days.

**Comment by auanasgheps** [comment]
It keeps happening. 
**A lot** since the last weeks.

**Comment by xd19981998** [comment]
I too am facing this issue very frequently since the past few weeks , closing the app and reopening it seems to work but it's pretty annoying .

**Comment by mavaa** [comment]
Experienced this very recently as well. My issue was (possibly) related to having the file in nextcloud. Every time the file changed from an external machine I would get this issue. Tried today to force stop KeePassDX, Clear data & cache and open the app again. After that it worked, and has done so for some hours now, even after changing my db file several times.

Not sure what caused this to begin with, but I'll report back if it starts again.

**Comment by soredake** [comment]
@Kunzisoft @J-Jamet @bpellin

**Comment by J-Jamet** [comment]
https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync#explanation

**Comment by soredake** [comment]
@J-Jamet so, there is nothing can be done from keepassdx side?

**Comment by horbo** [comment]
> https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync#explanation

I use Syncthing to sync db file among my devices - on the phone file is stored locally, so i don't have to use any file manager (beside one built-in in android), thus there is no way to file link to be broken.
If file would be broken, reselecting the database should help, but it doesn't. I have to force close the app to start work again :(

**Comment by soredake** [comment]
As temporary alternative there is https://github.com/PhilippC/keepass2android and https://github.com/authpass/authpass which support gdrive natively bypassing SAF.

**Comment by J-Jamet** [comment]
> Experienced this very recently as well. My issue was (possibly) related to having the file in nextcloud. Every time the file changed from an external machine I would get this issue. Tried today to force stop KeePassDX, Clear data & cache and open the app again. After that it worked, and has done so for some hours now, even after changing my db file several times.

@mavaa It is not normally necessary to make a clear cache. Here it worked for you simply because you reselected the file after the clear which generated a new URI from NextCloud. Re-selecting the file from Nextcloud would have been the same.

> so, there is nothing can be done from keepassdx side?

@soredake It is possible to improve the reading of the stream (what I have already done and don't see how to improve), but not all the problems you describe are due to a stream reading.
Here it is simply that the URI delivered by the third app that accesses the file does not provide a stream anymore or changes the metadata of this URI. I repeat it but it is as if the internet address (URL) changed (the server changed its configuration), you may have a bookmark on your browser (here corresponds to the history of KeePassDX), if the address pointed to is no longer the same, there is nothing to do by the client.

> I use Syncthing to sync db file among my devices - on the phone file is stored locally, so i don't have to use any file manager (beside one built-in in android), thus there is no way to file link to be broken.

@horbo You can't use KeePassDX without a file manager, as you say, there is one by default in your system otherwise you won't even be able to read a file if the URI is not linked to a data stream.

> If file would be broken, reselecting the database should help, but it doesn't. I have to force close the app to start work again :(

This case is very strange, it is not necessary to kill KeePassDX. Basically the URI flow is initialized when the activity is created so closing KeePassDX normally and reopening it should reset the flow initialization. After all it depends on your OS, many Android devices replace the File system of the AOSP with a proprietary file manager, this is the case of Samsung, Xiaomi, Huawei and many others. Can you tell me what is the package name of your file manager and its version?

But again, you should see the KeePassDX URI history only as a help. It's like opening your text file editor on your PC and viewing recent files. In 90% of the cases, users use their PC's file manager to access a file and do not use the recent files editor. They know that it may become inaccessible to the app editor, why does it become a huge problem to select it from its location when you use a phone? 
Do you think I should make this history less accessible or just hide it by default so that users mainly use the open button or go through their file manager to select a file?

**Comment by horbo** [comment]
@J-Jamet 

> You can't use KeePassDX without a file manager, as you say, there is one by default in your system otherwise you won't even be able to read a file if the URI is not linked to a data stream.

Yes, I meant that I am not using any custom file managers for KeePass, especially those that store files in the cloud (Google Drive, One Drive etc.).
What I wanted to point out is, I store kdbx file locally on the phone memory, and use default file manager to select the file in KeePass. Can't find an appId of the default file manager though. I'm using Galaxy S9+, and when selecting a file, KeePass uses default file selector.

As per killing the app. When problem occurs, **I can even delete a database file from recent list, and select it once again using file selector**, the message `Access to the file revoked by the file manager` still remains.

**Comment by JourPT** [comment]
I had this error minutes ago, without my knowledge my SD card (where I store my database file) had for a strange reason unmounted. After I rebooted and the card correctly mounted, everything was fine again. 
I know it might not be the case for everyone but I got this exact error and this how I solved it.
Best regards.

**Comment by J-Jamet** [comment]
@horbo You can easily see the package of an application with https://github.com/MajeurAndroid/android-applications-info.
You have a Galaxy S, I think your default manager is the one from Samsung (second to last in the list https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync)

> I can even delete a database file from recent list

That's strange, it means that the manager ignores the delete requests. If you show me the package with the tool mentioned above, I can investigate.

**Comment by J-Jamet** [comment]
@JourPT In your case it is normal, thanks for the feedback.

**Comment by TakeshiKovacs999** [comment]
I have this issue as well. I have tried both local and on google drive. Seems to be a path and permission issue. I wonder if google changed a rule in file manager that is causing this issue... I just have to reselect the file which must fix the path as well as correcting the permission issue.  I'm just running a stock pixel 4a on Android 11. Perhaps but I'm only guessing accessing log files not really a thing on stock Android. Perhaps this will help.
https://source.android.com/security/bulletin/2021-06-01

**Comment by alexanderadam** [comment]
Hm, now I'm also running into this.
When I had [a similar issue the last time](https://github.com/Kunzisoft/KeePassDX/issues/501) I was asked by the system which file manager to use (I have Material File installed).
But now the system doesn't ask.

Any idea how to enforce this?

**Comment by J-Jamet** [comment]
It is possible that an application is set as default to "OPEN_DOCUMENT" intent. It must be possible to change it in the settings of your device but I do not know where.

**Comment by clemdumo** [comment]
Hi guys, I thought I'd chime in. For a few days I've had a similar issue : 

* if I try to open KeepassDX from a field prompt (i.e. to log into a site or app), the screen will display an "access revoked by the file manager" error. Also the name at the top of the sign-in screen is all weirded out (looks like part of the URI instead of the actual filename) and while I'm still being prompted for credentials, that fails with a Javascript security exception (permission denied, which sounds logical in that situation) ; if thereafter I try to open the app manually, the problem is identical, but I can just hit the back button and reselect the file from the recent files list (credentials are still valid and will grant access), 

* as long as I don't try to open KeepassDX from a field prompt but only open the app manually, it works as expected, with the filename correctly displayed and no error, until the next time I try to open it from a field prompt. 

So I'm not sure KeepassDX is the culprit here but any pointers or thoughts could help.

I'm using KeepassDX Pro 2.10.5 on a Pixel 5 with Stock Android and the default file manager. Of course I can add details I might not have thought of if you need.

**Comment by J-Jamet** [comment]
I tried to improve the URI permission detection system in version 3.0.0. You will tell me if it is better for you or not when it is in beta.

**Comment by alexanderadam** [comment]
Can I install the beta from F-Droid?

**Comment by J-Jamet** [comment]
No, at the moment I haven't even finished the development. The beta will be on the Play Store. Otherwise it is possible to compile the source code.

**Comment by clemdumo** [comment]
> I tried to improve the URI permission detection system in version 3.0.0. You will tell me if it is better for you or not when it is in beta.

Cheers, I'm looking forward to downloading the beta once it's out :)

**Comment by J-Jamet** [comment]
Also don't expect broken links to work on their own again if the file manager doesn't want, but I think it will solve a bug where some users had to go back to the database list to reset the link validity in the login screen.

**Comment by J-Jamet** [comment]
3.0.0_beta01 is out

**Comment by clemdumo** [comment]
Cheers. It just got pushed now and after testing relatively to the issue I described above : 

— when entering selection mode from a field prompt, it would try to open the latest open database from an incorrect URI : now it sends me to the database selection screen, allowing me to pick which database to open which prevents the error from occuring, 

— when opening the app manually after having attempted to from a field prompt,  the same error would occur but *not* it the app was manually open and the database correctly loaded the previous time : obviously that is fixed since the error doesn't happen ln the first place. 

I don't know whether that was the intent, but that definitely works for me. 

So well done thanks for that 👍🏻 👍🏻 👍🏻

**Comment by auanasgheps** [comment]
Hey, I have a _related_ (maybe?) issue to the revoked file access. Still using the latest stable release.

When I open the app, sometimes, I can't see any DB at all, without any error message. However, If I force kill and then reopen the app, I'm prompted for the biometric unlock of my DB which opens and syncs fine.
So in this case the access is not revoked by the file manager, but it's temporary _lost_ by KPDX. I've seen this issue occurring a couple of times in the last few days but I have no idea how to replicate it.

**Comment by J-Jamet** [comment]
@auanasgheps I think in your case the loading of metadata from database files is not done correctly. I don't really know what can cause this, maybe the content provider that is supposed to deliver the information of a URI doesn't initialize, but in this case, I can't do much. The new version 3.0.0 improves the linking of URIs, you tell me if it's better.

**Comment by godSaysHODL** [comment]
The newest release is messing up file permissions. I have to delete and restore the configuration for my key database everytime I want to unlock it. I cannot read the full error either because of UI constraints.

This was not happening in previous releases

**Comment by J-Jamet** [comment]
@godSaysHODL This issue is fixed since version 3.0.0_beta02 #1066 #1067 #1071

**Comment by godSaysHODL** [comment]
> @godSaysHODL This issue is fixed since version 3.0.0_beta02 #1066 #1067 #1071

Okay thanks. I m using beta01 from the Fdroid store

**Comment by clemdumo** [comment]
Beta 02 seems to not remember the latest databases. When opening the app manually, it doesn't prompt the latest file to open. I have to go to Open existing database, select my database again, then it remembers the credentials so I don't have to load the key file again or enter my password if I had the biometric credential enabled. 

Minor inconvenience that doesn't really cause problems, but I thought i'd mention it.

**Comment by J-Jamet** [comment]
@clemdumo I don't have this problem, it must depend on your file manager which invalidates the links. If you deactivate the option "Settings - App settings - Hide broken database links", you should see your old link but not be able to open it anymore.

**Comment by godSaysHODL** [comment]
Is there an APK for beta 2 or 3 floating around? Or maybe ETA to Fdroid? I'm not actually developer enough to compile these things from source 😅

**Comment by J-Jamet** [comment]
Currently not because they are only TAGs on the develop branch for testing. I didn't even think that making TAGs on this develop branch would make the compilation automatic on F-Droid. But as it is the case, the next beta versions will also be built automatically.

**Comment by clemdumo** [comment]
@J-Jamet Mmh, no it doesn't look like the link is broken since once the database file is re-selected it remembers everything and the credentials are still valid. It just that the list of last opened databases is sometimes empty (I tried to reproduce and it seems random, I'll keep investigating).

**Comment by J-Jamet** [comment]
@clemdumo OK, keep us informed. :)

**Comment by clemdumo** [comment]
> @clemdumo OK, keep us informed. :)

Looks like you were partly right. I disabled "mask broken URIs" from the settings. When it happens I now see the latest database appear in the form of a URI. 

… except though it looks broken, it isn't as clicking on it does prompt for credentials (memorised credentials i.e.finger print and key file are still valid) and open the correct database. 

That's a weird combo.

Can't be sure of what causes this though, it seems it only happens after I haven't opened it for a while. Could be related to the file manager then I guess.

**Comment by J-Jamet** [comment]
@clemdumo I see what's going on. Concretely, from version 3.0.0 (beta), there is a check at the time of the creation of the list and at the time of the loading of the password screen (to check the read and write rights to the content provider of the file manager).
When the list is displayed, the file manager you are using has to say "No, I have revoked the URI you are requesting to read and write", ok, so KeePassDX hides it (this is done by looking at the metadata of the file which is not available).
But then on the password screen, the content provider provides the data stream even though it said it was not available.
So I can't do much about that if the content provider has inconsistent behavior.
In the previous version of KeePassDX, there was no check, so it didn't matter if the authorization was good or not, you knew it when you asked for the metadata or the stream. And users were complaining because they couldn't open their database (with the "Access to the file revoked by the file manager message"), so I added the check system (But of course, I assume that content providers do what they are supposed to do :D) .
For you, which file manager produces the faulty URI?

**Comment by clemdumo** [comment]
Oh I see. That makes sense.

I'm using Google's default file manager on my Pixel 5 so that's that. The issue is pretty new though. Maybe *they* changed something. 

Right now it still works as opposed to before so I'm still happy.

**Comment by rogers1106** [comment]
Same problem here with LOS 18.1 (Android 11) and v3.0.0. Filemanager in use is Files.
For file operations I use Total Commander and the database is synced with Syncthing-Fork.

**Comment by rogers1106** [comment]
It happens independent from Syncthing (closed). Sometimes immediately after closing KeyPass, sometimes after multiple closes.

**Comment by J-Jamet** [comment]
I don't understand, how can you see the status of database URIs from KeePassDX if you close it? The application has to be open if you want to see these states. Are you talking about the KeePass application for PC?

**Comment by rogers1106** [comment]
I can see that KeePass has lost the information about the last opened file when I restart KeePass.

**Comment by J-Jamet** [comment]
OK, well I can't reproduce this behavior at all with my Android 11 emulator and default file manager.
So it's necessary to look at the source code of the LOS 18.1 Files to see when it invalidates the link and why `takePersistableUriPermission` doesn't work.
I add a note in the wiki

**Comment by rogers1106** [comment]
It's the same with LOS 16.0 Files app. Just tested.

**Comment by soredake** [comment]
>los 18.1

It's been a problem in 17.1 too.

>or only once in a while? 

It's happened reliable after week or two.

**Comment by J-Jamet** [comment]
I also have Lineage OS 17.1 on one of my devices and I have no problem.
@soredake You said you were using a URI from the Play Store, for me it's normal if it hasn't finished synchronizing or something like that, it must be a protection, I've only briefly tested the Play Store integration in the Files app.
But locally I'm sure it works. (URI types `content://com.android.externalstorage.documents/document/...`)
Can you indicate the type of URI for which it does not work for you?

**Comment by soredake** [comment]
> Can you indicate the type of URI for which it does not work for you?

I'll do it when it breaks again.

**Comment by J-Jamet** [comment]
I repeat, if the URI is created by a content provider that belongs to a file manager that breaks links, I can't do anything from KeePassDX.
So those who are using the `File` application but are using `Google Drive`, it's the same problem, it's `Google Drive` that produces the URIs (type `content://com.google.android.apps.docs.storage/document/...`)  and therefore breaks the links for one reason or another.

You just have to reselect the file from its original location.
The solution if you want to keep a link constantly present in the history is to create a new file manager application that integrates the APIs of the Clouds by making a caching and file comparison system to provide a clear state of the files provided by the file provider.

**Comment by rogers1106** [comment]
This means that all LOS users will have the same problem?!

**Comment by J-Jamet** [comment]
This means that the behavior depends on the content provider of the file manager you use.
The File application of Lineage OS works very well (for me and based on all tests I did) in all versions but my guess is that you simply use the `Google Drive` application (via the link in the `File` application) to provide your URI.
I created the wiki on purpose to aggregate all the information on the behavior of several Cloud applications. https://github.com/Kunzisoft/KeePassDX/wiki/File-Manager-and-Sync

**Comment by rogers1106** [comment]
Using the Drive file manager would not lead me to the local file. That is what I know (not guess). Bye

**Comment by J-Jamet** [comment]
Which Drive file manager are you talking about? Can you point out the URI you are using that is causing the problem?

**Comment by jvc2015** [comment]
Finally i don't can find the database in all directores. I actually use other open software (AuthPass) since months and no problems.

Thanks very much for all you request to my answer!

**Comment by rogers1106** [comment]
You guessed that I use the Google Drive file manager and I tried to explain you that I do not use it because it cannot lead me to local files.

I do use the LOS Files app where I choose Pixel4a, then the folder that contains my databases, then the database of choice. That's all.

If I find a reason for the problem I will come back and report here.

Thanks and bye

**Comment by J-Jamet** [comment]
@jvc2015 I understand that the number of comments and notifications is no longer useful to you. Thank you fro your feedback.

I'm opening a separate discussion channel to continue the thread. https://github.com/Kunzisoft/KeePassDX/discussions/1089
